crypto/evp/digest.c

```diff
@@ -629,6 +629,20 @@ int EVP_Digest(const void *data, size_t count,
     return ret;
 }
 
+int EVP_Q_digest(const void *data, size_t count,
+                 unsigned char *md, unsigned int *size,
+                 OSSL_LIB_CTX *libctx, const char *name, const char *propq)
```

Comment:
Can we be consistent at least with the _Q functions in the other PR's and have the libctx and propq in the same place :)

Reply:
> Can we be consistent at least with the _Q functions in the other PR's and have the libctx and propq in the same place :)

The libctx and propq are placed around the digest alg name, as done also for the mac alg name of `EVP_Q_mac()`:
```
unsigned char *EVP_Q_mac(OSSL_LIB_CTX *libctx, const char *name, const char *propq,
                         const char *subalg, const OSSL_PARAM *params, ...
```
The only discrepancy is that for `EVP_Q_digest()` the alg name is at the end.
In this regard I followed the model of `EVP_Digest()`:
```
__owur int EVP_Digest(const void *data, size_t count,
                          unsigned char *md, unsigned int *size,
                          const EVP_MD *type, ENGINE *impl);
```
so IMHO the arg order is a consistent with related functions as it can be.

Reply:
IMO there is no reason to follow the EVP_Digest argument order here. I'd make it as much as possible consistent with EVP_Q_mac and disregard the EVP_Digest.

Reply:
Done